src/main/java/org/opensearch/knn/index/query/rescore/RescoreContext.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.query.rescore;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.opensearch.common.ValidationException;
+
+@Slf4j
+@Getter
+@AllArgsConstructor
+@Builder
+@EqualsAndHashCode
+public final class RescoreContext {
+
+    public static final float DEFAULT_OVERSAMPLE_FACTOR = 1.0f;
+    public static final float MAX_OVERSAMPLE_FACTOR = 5.0f;
```

Comment:
@jmazanec15 will 5 be good enough may be 100? I know nothing is good enough. but I was thinking with 100 as max and a cap of 10K on the k*oversampling factor should cover most of the cases.

Thoughts?

Reply:
Sure I can increase. Let me update